### PR TITLE
An starting on UI

### PR DESCRIPTION
**Pull request by AdolfoNava**

### CODE REVIEW

**Review by brvarner — APPROVED** [review]
Looks good (this time it's for real!)

**Review by MayaS1111 — APPROVED** [review]
Looking gooooood

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,2 +1,3 @@
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
Comment (JoseMDO) [inline app/controllers/application_controller.rb:2]: Nice!

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -5,12 +5,24 @@
     <meta name="viewport" content="width=device-width,initial-scale=1">
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
-
+    <%= render partial: 'shared/cdn_assets' %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:8]: Great job adding your CDN assets to a partial as this helps clean up the application layout file :)

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -5,12 +5,24 @@
...
     <%= stylesheet_link_tag "application", "data-turbo-track": "reload" %>
     <%= javascript_importmap_tags %>
   </head>
 
   <body>
-    <%= yield %>
+    <%= render partial: "shared/navbar" %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:14]: same as above

File: app/views/layouts/application.html.erb [hunk 4]
```diff
@@ -5,12 +5,24 @@
...
+    <div class="container">
+      
+      <% if notice.present? %>
+        <%= render partial: "shared/flash", locals: {message: notice, css_class: "success" } %>
```
Comment (JoseMDO) [inline app/views/layouts/application.html.erb:18]: same as above

File: app/views/shared/_navbar.html.erb [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+      <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+      <ul class="navbar-nav mr-auto">
+        <li class="nav-item dropdown">
+          <a class="nav-link dropdown-toggle" href="#" role="button" data-bs-toggle="dropdown" aria-expanded="false">
+            Resources
+          </a>
+          <div class="dropdown-menu">
+            <a class="dropdown-item" href="/photos">Photos</a>
```
Comment (JoseMDO) [inline app/views/shared/_navbar.html.erb:14]: remember we can start using helper methods here like link_to

File: lib/tasks/dev.rake [hunk 6]
```diff
@@ -8,6 +8,20 @@ task({ :sample_data => :environment }) do
     Photo.destroy_all
     User.destroy_all
   end
+  usernames = Array.new { Faker::Name.first_name }
+
+  usernames << "alice"
+  usernames << "bob"
+
+  usernames.each do |username|
```
Comment (JoseMDO) [inline lib/tasks/dev.rake:16]: great job on sample data!

**Review by JoseMDO — APPROVED** [review]
Great job Adolfo!

**Comment by FadiBaker92** [comment]
Looks great Adolfo! Just a tiny note: try to use the link_to helper method instead of the anchor tags.

**Pull request merged**